### DISCUSSION TITLE
JSON schema does not compile

### DISCUSSION THREAD

**Discussion by alephnull** [goreleaser/goreleaser#4510] [Q&A] [answered]
When attempting to use github.com/santhosh-tekuri/jsonschema/v5 to validate config files, I am unable to compile https://goreleaser.com/static/schema.json due to the errors:
```
panic: jsonschema: jsonschema file:///home/alok/work/tyk/src/gromit/gorel.json compilation failed: "/$defs/File/oneOf/1" and "/$defs/NixDependency/oneOf/1" in https://github.com/goreleaser/goreleaser/pkg/config/project have same canonical-uri
panic: jsonschema: jsonschema file:///home/alok/work/tyk/src/gromit/gorel.json compilation failed: "/$defs/File/oneOf/1" and "/$defs/Hooks/oneOf/1/items/oneOf/1" in https://github.com/goreleaser/goreleaser/pkg/config/project have same canonical-uri
```

This is my first attempt at using json schemas so I tried an online validator but they either do not support draft 2020-12 or fail to parse the schema. I'm trying to figure out where the problem is. I have used the library to validate against the Github Actions json schema but that is draft-07.

**Comment by caarlos0** [comment, marked as answer]
should be fixed, see #4511 and https://github.com/goreleaser/goreleaser/commit/9ce21d3b95ca2ca894bcd96d8f4731951fee0ef1

**Comment by alephnull** [comment]
Thanks! Happy 2024!
